### PR TITLE
Improve config file loading

### PR DESCRIPTION
**Pull request by brianhelba**
This may be easier to review looking at individual commits.

### CODE REVIEW

**Comment by brianhelba** [comment]
I may add some additional commits to this.

**Comment by brianhelba** [comment]
@joanne-tsai @jbeezley Ready for review. PTAL.

File: test/test_config.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
+from tempfile import NamedTemporaryFile
+
+from simulation.config import SimulationConfig
+
+
+def test_config_defaults():
+    config = SimulationConfig()
+    assert config.getboolean('simulation', 'validate') is True
```
Comment (joanne-tsai) [inline test/test_config.py:12]: I read about getboolean online and I don't quite understand how it works and what is going on here.

File: simulation/config.py [hunk 2]
```diff
@@ -43,6 +43,8 @@ def __init__(self, *config_sources: Union[str, PurePath, dict]) -> None:
         for config_source in config_sources:
             if isinstance(config_source, dict):
                 self.read_dict(config_source)
+            if isinstance(config_source, TextIOBase):
```
Comment (joanne-tsai) [inline simulation/config.py:47]: why is this not elif?

Comment (brianhelba) [reply simulation/config.py:47]: It should be, thanks.

**Pull request merged**